### PR TITLE
Hunterisation of thrift

### PR DESCRIPTION
**Pull request by JochenGuck**
I applied the changes done on thrift version 0.11 on the thrift version 0.12 branch and update to the latest hunter gate.

I tested the changes as described in:
https://docs.hunter.sh/en/latest/creating-new/update.html

thrift is successfully compiling the THRIFT_ROOT is set correctly. 

However cmake find_package is not able to find the package configuration file.

is there something wrong in this package or do I miss some changes in the hunter config?

### CODE REVIEW

**Comment by ruslo** [comment]
> is there something wrong in this package or do I miss some changes in the hunter config?

Since you're not providing any links to CI jobs or log, it's really hard to say.

**Comment by JochenGuck** [comment]
I checked the packet structure and indeed the thriftConfig.cmake file is missing. 
There is no cmake folder under the install/lib folder at all. 
And the boost library are also not copied.

File: lib/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -214,6 +223,86 @@ install(DIRECTORY "src/thrift" DESTINATION "${INCLUDE_INSTALL_DIR}"
...
+    DIRECTORY "src/thrift"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+    FILES_MATCHING PATTERN "*.h"
+)
+
+# Config
+#   * <prefix>/lib/cmake/thrift/thriftConfig.cmake
+#   * <prefix>/lib/cmake/thrift/thriftConfigVersion.cmake
+install(
+    FILES "${project_config}" "${version_config}"
```
Comment (ruslo) [inline lib/cpp/CMakeLists.txt:294]: > I checked the packet structure and indeed the thriftConfig.cmake file is missing.

That's the place where this file is installed.

File: build/cmake/DefineOptions.cmake [hunk 2]
```diff
@@ -114,7 +104,11 @@ if(BUILD_CPP)
         list(APPEND boost_components unit_test_framework filesystem chrono program_options)
     endif()
     if(boost_components)
-        find_package(Boost 1.53 REQUIRED COMPONENTS ${boost_components})
+        hunter_add_package(Boost COMPONENTS ${boost_components})
```
Comment (ruslo) [inline build/cmake/DefineOptions.cmake:107]: >  And the boost library are also not copied.

To clarify, Boost should be installed to the Hunter root folder. That is the command to trigger Boost build.

**Review by ruslo — CHANGES_REQUESTED** [review]
> I tested the changes as described in:

> thrift is successfully compiling

Can you show the exact steps you're doing to test the code?

**Comment by JochenGuck** [comment]
Thanks to you're suggestions i reevaluated my test configuration. 
I figured out that i copied the wrong URL to hunter.cmake which pointed to the original thrift version.

By correcting this mistake I was able to compile the thrift 0.12 library with a visual studio 2017 compiler.

**Comment by ruslo** [comment]
> By correcting this mistake I was able to compile the thrift 0.12 library with a visual studio 2017 compiler.

👍 

Thanks, released:
- https://github.com/hunter-packages/thrift/releases/tag/v0.12.0-p0

For confirming that the rest of the toolchains are working and submitting a new version to Hunter, please check the same documentation page:
- https://docs.hunter.sh/en/latest/creating-new/update.html

**Pull request merged**